test/e2e/multikueue/e2e_test.go

```diff
@@ -1044,6 +1044,123 @@ var _ = ginkgo.Describe("MultiKueue", func() {
+			})
+
+			ginkgo.By("Verifying both jobs are admitted (proving separate quotas)", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdJobMk := &batchv1.Job{}
+					createdJobRegular := &batchv1.Job{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(jobMk), createdJobMk)).To(gomega.Succeed())
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(jobRegular), createdJobRegular)).To(gomega.Succeed())
+
+					// Both unsuspended = both admitted despite using full quotas
+					g.Expect(ptr.Deref(createdJobMk.Spec.Suspend, false)).To(gomega.BeFalse())
+					g.Expect(ptr.Deref(createdJobRegular.Spec.Suspend, false)).To(gomega.BeFalse())
+
+					g.Expect(createdJobMk.Status.StartTime).NotTo(gomega.BeNil())
+					g.Expect(createdJobMk.Status.Active).To(gomega.Equal(int32(1)))
+					g.Expect(createdJobMk.Status.CompletionTime).To(gomega.BeNil())
+
+					g.Expect(createdJobRegular.Status.StartTime).NotTo(gomega.BeNil())
+					g.Expect(createdJobRegular.Status.Active).To(gomega.Equal(int32(1)))
+					g.Expect(createdJobRegular.Status.CompletionTime).To(gomega.BeNil())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Finishing the MK job's pod", func() {
+				listOpts := util.GetListOptsFromLabel(fmt.Sprintf("batch.kubernetes.io/job-name=%s", jobMk.Name))
+				util.WaitForActivePodsAndTerminate(ctx, k8sWorker2Client, worker2RestClient, worker2Cfg, jobMk.Namespace, 1, 0, listOpts)
+			})
+
+			ginkgo.By("Finishing the regular job's pod", func() {
+				listOpts := util.GetListOptsFromLabel(fmt.Sprintf("batch.kubernetes.io/job-name=%s", jobRegular.Name))
+				util.WaitForActivePodsAndTerminate(ctx, k8sManagerClient, managerRestClient, managerCfg, jobRegular.Namespace, 1, 0, listOpts)
+			})
+
+			ginkgo.By("Waiting for both jobs to complete", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdJobMk := &batchv1.Job{}
+					createdJobRegular := &batchv1.Job{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(jobMk), createdJobMk)).To(gomega.Succeed())
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(jobRegular), createdJobRegular)).To(gomega.Succeed())
+
+					g.Expect(createdJobMk.Status.Conditions).To(gomega.ContainElement(gomega.BeComparableTo(
+						batchv1.JobCondition{
+							Type:   batchv1.JobComplete,
+							Status: corev1.ConditionTrue,
+						},
+						cmpopts.IgnoreFields(batchv1.JobCondition{}, "LastTransitionTime", "LastProbeTime", "Reason", "Message"))))
+					g.Expect(createdJobRegular.Status.Conditions).To(gomega.ContainElement(gomega.BeComparableTo(
+						batchv1.JobCondition{
+							Type:   batchv1.JobComplete,
+							Status: corev1.ConditionTrue,
+						},
+						cmpopts.IgnoreFields(batchv1.JobCondition{}, "LastTransitionTime", "LastProbeTime", "Reason", "Message"))))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+		})
```

Comment:
```suggestion
			jobs := []*batchv1.Job{jobMk, jobRegular}
			for _, job := range jobs {
    ginkgo.By(fmt.Sprintf("Waiting for %s job to complete", client.ObjectKeyFromObject(job)), func() {
       gomega.Eventually(func(g gomega.Gomega) {
          createdJob := &batchv1.Job{}
          g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(job), createdJob)).To(gomega.Succeed())

          g.Expect(createdJob.Status.Conditions).To(gomega.ContainElement(gomega.BeComparableTo(
             batchv1.JobCondition{
                Type:   batchv1.JobComplete,
                Status: corev1.ConditionTrue,
             },
             cmpopts.IgnoreFields(batchv1.JobCondition{}, "LastTransitionTime", "LastProbeTime", "Reason", "Message"))),
          )
       }, util.LongTimeout, util.Interval).Should(gomega.Succeed())
    })
}
```
I propose to go on this way. 

Or even better to create helper ExpectJobCompleted(). We can reuse it in other places.